src/app/clusters/boolean-state-configuration-server/boolean-state-configuration-cluster.cpp

```diff
@@ -16,447 +15,344 @@
+        if (mFeatures.Has(Feature::kAlarmSuppress))
+        {
+            ReturnErrorOnFailure(builder.AppendElements({ Commands::SuppressAlarm::kMetadataEntry }));
+        }
     }
+
     return CHIP_NO_ERROR;
 }
 
-CHIP_ERROR BooleanStateConfigAttrAccess::Write(const ConcreteDataAttributePath & aPath, AttributeValueDecoder & aDecoder)
+std::optional<DataModel::ActionReturnStatus>
+BooleanStateConfigurationCluster::InvokeCommand(const DataModel::InvokeRequest & request, chip::TLV::TLVReader & input_arguments,
+                                                CommandHandler * handler)
 {
-    if (aPath.mClusterId != BooleanStateConfiguration::Id)
+    switch (request.path.mCommandId)
     {
-        return CHIP_ERROR_INVALID_PATH_LIST;
-    }
+    case Commands::SuppressAlarm::Id: {
+        Commands::SuppressAlarm::DecodableType request_data;
+        ReturnErrorOnFailure(request_data.Decode(input_arguments));
 
-    switch (aPath.mAttributeId)
-    {
-    case CurrentSensitivityLevel::Id: {
-        return WriteCurrentSensitivityLevel(aPath, aDecoder);
+        return SuppressAlarms(request_data.alarmsToSuppress);
     }
-    default: {
-        break;
+    case Commands::EnableDisableAlarm::Id: {
+        Commands::EnableDisableAlarm::DecodableType request_data;
+        ReturnErrorOnFailure(request_data.Decode(input_arguments));
+
+        const auto alarms = request_data.alarmsToEnableDisable;
+
+        VerifyOrReturnError(mAlarmsSupported.HasAll(alarms), Status::ConstraintError);
+
+        if (mAlarmsEnabled != alarms)
+        {
+            mAlarmsEnabled = alarms;
+            NotifyAttributeChanged(AlarmsEnabled::Id);
+        }
+
+        if (mDelegate != nullptr)
+        {
+            // TODO: For backwards compatibility with previous code, we ignore the return code
+            //       from the delegate handler. This feels off though...
+            (void) mDelegate->HandleEnableDisableAlarms(alarms);
+        }
+
+        // This inverts the bits (0x03 is the current max bitmap):
+        //   - every "known" bit that is set to 0 in the request will be set to 1 in the `alarmsToDisable`
+        const BitMask<BooleanStateConfiguration::AlarmModeBitmap> alarmsToDisable{ static_cast<uint8_t>(~alarms.Raw() &
+                                                                                                        kAllKnownAlarmModes) };
+
+        bool emit = false;
+        if (mAlarmsActive.HasAny(alarmsToDisable))
+        {
+            mAlarmsActive.Clear(alarmsToDisable);
+            NotifyAttributeChanged(AlarmsActive::Id);
+            emit = true;
+        }
+        if (mAlarmsSuppressed.HasAny(alarmsToDisable))
+        {
+            mAlarmsSuppressed.Clear(alarmsToDisable);
+            NotifyAttributeChanged(AlarmsSuppressed::Id);
+            emit = true;
+        }
+
+        if (emit)
+        {
+            EmitAlarmsStateChangedEvent();
```

Comment:
If we are renaming it anyway, it should be called GenerateAlarmsStateChangedEvent, and the boolean guard should be called "generateEvent".